src/charm.py

```diff
@@ -74,6 +75,16 @@ def __init__(self, *args: typing.Any) -> None:
         self.framework.observe(
             self.on.secret_storage_relation_changed, self._on_secret_storage_relation_changed
         )
+        self._require_nginx_route()
+
+    def _require_nginx_route(self) -> None:
+        """Set up the requirer side of the nginx-route relation."""
+        require_nginx_route(
+            charm=self,
+            service_name=self.unit.name,
+            service_hostname=self.unit.name,
```

Comment:
I think you'd want `self.app.name` for both of these rather than `self.unit.name` as that would then be different for each unit.